contracts/core_amm/liquidity_pool.cairo

```diff
@@ -700,23 +698,33 @@ func expire_option_token_for_pool{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*
     local term = terminal_price;
     with_attr error_message("unable to split_option_locked_capital in expire_option_token_for_pool optsize {optsize}, optsize64 {optisize64} strike {str} term {term}"){
     let (long_value, short_value)  = split_option_locked_capital(
-        option_type, option_side, option_size_m64x61, strike_price, terminal_price
-    );
+            option_type, option_side, option_size_m64x61, strike_price, terminal_price
+        );
     }
 
     // Adjusts only the lpool_balance and pool_locked_capital storage_vars
     with_attr error_message("unable to adjust_lpool_balance_and_pool_locked_capital_expired_options in expire_option_token_for_pool"){
-    adjust_lpool_balance_and_pool_locked_capital_expired_options(
-        lptoken_address=lptoken_address,
-        long_value=long_value,
-        short_value=short_value,
-        option_size=option_size,
-        option_side=option_side,
-        maturity=maturity,
-        strike_price=strike_price
-    );
+        adjust_lpool_balance_and_pool_locked_capital_expired_options(
+            lptoken_address=lptoken_address,
+            long_value=long_value,
+            short_value=short_value,
+            option_size=option_size,
+            option_side=option_side,
+            maturity=maturity,
+            strike_price=strike_price
+        );
     }
 
+    let (current_pool_position) = get_option_position(
+        lptoken_address, option_side, maturity, strike_price
+    );
+
+    let new_pool_position = current_pool_position - option_size;
+    with_attr error_message("new_pool_position is negative in expire_option_token_for_pool") {
+        assert_nn(new_pool_position);
```

Comment:
can we make this assert_le? this shouldn't be a problem with felts, but just to be overly cautious?

Reply:
you mean assert_le(0, stuff)?

Reply:
assert_le(option_size, current_pool_position)